pyomo/contrib/solver/common/persistent.py

```diff
@@ -10,21 +10,37 @@
 #  __________________________________________________________________________
 
 import abc
+import datetime
 from typing import List
 
 from pyomo.core.base.constraint import ConstraintData, Constraint
 from pyomo.core.base.sos import SOSConstraintData, SOSConstraint
 from pyomo.core.base.var import VarData
 from pyomo.core.base.param import ParamData, Param
 from pyomo.core.base.objective import ObjectiveData
+from pyomo.core.staleflag import StaleFlagManager
 from pyomo.common.collections import ComponentMap
 from pyomo.common.timing import HierarchicalTimer
-from pyomo.core.expr.numvalue import NumericConstant
-from pyomo.contrib.solver.util import collect_vars_and_named_exprs, get_objective
+from pyomo.contrib.solver.common.results import Results
+from pyomo.contrib.solver.common.util import collect_vars_and_named_exprs, get_objective
 
 
 class PersistentSolverUtils(abc.ABC):
```

Comment:
NOT TODO for this PR:

- What is the intent of this class?
  - To manage automatic checking of state / checking differences in the model and updating them; implements the "Auto" part of APPSI
  - What is a better name? `PersistentSolverStateTracker`? This is the classic "is-a" vs. "has-a". Other ideas: `PersistentSolverChangeDetector`, `PersistentModelStateManager`, `PersistentSolverUpdater`

**Summary**: We don't want the persistent solver interfaces to inherit from this; we want the classes to be able to contain one of these classes and use it. `ModelChangeObserver`

Reply:
Also probably won't need the `_func_name` and `func_name` standard here when re-worked.